File: java/src/main/java/com/example/demo/repositories/GreetingRepository.java
package com.example.demo.repositories;

import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Optional;
import java.util.stream.Collectors;

import com.example.demo.entities.Greeting;

public class GreetingRepository implements IGreetingRepository {

    private final Map<Long,Greeting> greetingMap;
    private Long autoIncrement = 1L;

    public GreetingRepository(){
        greetingMap = new HashMap<Long,Greeting>();
    }

    @Override
    public Greeting save(Greeting greeting) {
        Greeting g = new Greeting(greeting.getMessage(),autoIncrement);
        greetingMap.put(autoIncrement,g);
        ++autoIncrement;
        return g;
    }

    @Override
    public boolean existsById(Long id) {
        return greetingMap.containsKey(id);
    }

    @Override
    public Optional<Greeting> findById(Long id) {
        return Optional.ofNullable(greetingMap.get(id));
    }

    @Override
    public List<Greeting> findAll() {
        return greetingMap.values().stream().collect(Collectors.toList());
    }

    @Override
    public void deleteById(Long id) {
        greetingMap.remove(id);
    }

    @Override
    public long count() {
        return greetingMap.size();
    }
    
}

File: python/src/repositories/GreetingRepository.py
from typing import List, Optional

from src.entities import Greeting


class GreetingRepository:
    def __init__(self) -> None:
        self._greetingMap = {}
        self._autoIncrement = 1

    def save(self,greeting: Greeting) -> Greeting:
        g = Greeting(greeting.get_message(),self._autoIncrement)
        self._greetingMap[self._autoIncrement] = g
        self._autoIncrement += 1
        return g

    def existsById(self,id: int) -> bool:
        if id in self._greetingMap:
            return True
        else:
            return False

    def findById(self, id: int) -> Optional[Greeting]:
        return self._greetingMap.get(id)

    def findAll(self) -> List[Greeting]:
        return list(self._greetingMap.values())

    def deleteById(self,id: int) -> None:
        del self._greetingMap[id]

    def count(self) -> int:
        return len(self._greetingMap)